compiler/src/main/java/com/juul/pommel/compiler/PommelWriter.kt

```diff
@@ -44,28 +45,48 @@ class PommelWriter(
                 }
             }
             .addMethod(
-                MethodSpec.methodBuilder(targetType.rawClassName().provideFunctionName())
-                    .addAnnotation(PROVIDES)
-                    .apply { scope?.let { addAnnotation(it) } }
-                    .addModifiers(Modifier.PUBLIC)
-                    .returns(binds)
-                    .applyEach(parameters) {
-                        addParameter(it.qualifiedType, it.simpleName.toString())
-                    }
-                    .addStatement(
-                        "return new \$T(\n\$L)", targetType,
-                        parameters.map { CodeBlock.of("\$N", it.simpleName) }.joinToCode(",\n")
-                    )
-                    .build()
+                if (targetType == binds) {
+                    writeProvidesMethod()
+                } else {
+                    writeBindsMethod()
+                }
```

Comment:
Is this `if` flipped?

If it's not (and you did intend to `writeBindsMethod` when `targetType != binds` ) then you should probably add a comment _why_; as the logic is confusing with how the code reads.

Reply:
This does look a bit confusing but what it is essentially saying is that if the return type is equal to the class `@SoloModule` was annotated on then generated the provide method else we can assume the developer passed a class in the annotation like so `@SoloModule(SomeInterface::class)` in which case we can generate a bind method instead. maybe its confusing because of the field names. perhaps changing `binds` to `returnType`. would that make more sense?

Reply:
If the logic is correct then it's fine as-is. Just might warrant a small comment explaining the logic (i.e. an abbreviated version of what you just explained).

Reply:
added a comment to explain the logic